src/Data/Field/Galois/Prime.hs

```diff
@@ -61,9 +62,7 @@ instance KnownNat p => GaloisField (Prime p) where
 instance KnownNat p => Group (Prime p) where
   invert = recip
   {-# INLINE invert #-}
-  pow y@(P x) n
-    | n >= 0    = P $ powModNatural x (fromIntegral n) $ natVal (witness :: Prime p)
-    | otherwise = pow (recip y) $ P.negate n
+  pow x  = P . unMod . (^%) (unsafeCoerce x :: Mod p)
```

Comment:
In order to use the hidden constructor `Mod` indirectly, I had to add six instances of `unsafeCoerce`, which currently works but may cause some problems with more aggressive optimisations.